docs/advanced-usage.md

```diff
@@ -31,6 +31,36 @@ fail or not.
 If you don't specify a `penalty` configuration, the plugin will use the [default threshold values][penaltyextensioncode], which are to
 allow any warning, but break the build on any error.
 
+#### Default `penalty` profiles
+
+It's also possible to use one of the default `penalty` profiles instead of specifying thresholds explicitly as follows:
+
+```gradle
+staticAnalysis {
+    penalty none
+}
+```
+
+In this case the build won't fail no matter how many violations (warnings or errors) are found.
+
+```gradle
+staticAnalysis {
+    penalty failOnErrors
+}
+```
+
+Default policy. This will break the build if any error is found. Warnings instead are only logged and will not break the build.
+
+
+```gradle
+staticAnalysis {
+    penalty failOnWarnings
+}
+```
+
+Use this if you want to fail fast: the build will break if any warning or error is found.
```

Comment:
Given this is what it does then maybe we should rename `failOnWarnings` to `failFast`, it would seem much more clear as `failOnWarnings` doesn't clearly imply `failOnErrors` as a name IMO

Reply:
good point @rock3r, we can address in this PR I guess

Reply:
Good point, I also saw that in the legacy version of this tool, there was an enum with values `NONE`, `ERRORS` and `ALL`. `ALL` was originally `WARNINGS`. 

So doing this will align that.

Reply:
Although GH thinks that 05c9492 addressed this, it didn't — please don't lose track of this rename 💃

Reply:
We have this now. https://github.com/novoda/gradle-static-analysis-plugin/issues/87

I was thinking to just change this in this PR but that requires a release. We should do it seprately.